### PR TITLE
Format native code using clang-format

### PR DESCRIPTION
**Pull request by pellared**
## Why

Fixes https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/1356

## What

- Adds automation for formatting C++ code
- Format the C++ code
- Largely based on https://github.com/dotnet/runtime

## Tests

CI + manual locally

## Checklist

- [x] ~~`CHANGELOG.md` is updated.~~
- [x] Documentation is updated.
- [x] New features are covered by tests.

### CODE REVIEW

File: scripts/format-native.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+fi
+
+if [ "$OS_TYPE" == "windows"  ]; then
+    EXT=.exe
+else
+    EXT=
+fi
+
+# files to format
+NATIVE_FILES=$(find . -iname *.cpp -o -iname *.hpp -iname *.h -iname *.inc | grep -v ./packages | grep -v ./src/OpenTelemetry.AutoInstrumentation.Native/lib)
```
Comment (pellared) [inline scripts/format-native.sh:34]: This line is the main reason why it is not yet implemented in Nuke. I think we can create an issue to move the formatting logic to Nuke targets. Let me know what you think if we should address it ASAP or can we postpone it.

**Comment by pellared** [comment]
@open-telemetry/dotnet-maintainers I suggest merging https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2088 before this PR.

**Review by Kielek — APPROVED** [review]

File: src/OpenTelemetry.AutoInstrumentation.Native/integration.cpp [hunk 2]
```diff
@@ -10,171 +10,170 @@
...
+    , locale(GetLocaleFromAssemblyReferenceString(str))
+    , public_key(GetPublicKeyFromAssemblyReferenceString(str))
 {
 }
 
 AssemblyReference* AssemblyReference::GetFromCache(const WSTRING& str)
 {
     std::lock_guard<std::mutex> guard(m_assemblyReferenceCacheMutex);
-    auto findRes = m_assemblyReferenceCache.find(str);
+    auto                        findRes = m_assemblyReferenceCache.find(str);
```
Comment (zacharycmontoya) [inline src/OpenTelemetry.AutoInstrumentation.Native/integration.cpp:33]: This is odd, is there some misconfiguration?

Comment (zacharycmontoya) [reply src/OpenTelemetry.AutoInstrumentation.Native/integration.cpp:33]: I take that back, it looks like this variable name is being aligned with the variable name on the previous line

**Review by zacharycmontoya — APPROVED** [review]

**Pull request merged**